resource/config.go

```diff
@@ -372,3 +372,14 @@ func TransformAttributeMap[T any](attributes utils.AttributeMap) (T, error) {
 	}
 	return out, nil
 }
+
+// NewConfigValidationError returns a config validation error occurring at a given path.
+func NewConfigValidationError(path string, err error) error {
```

Comment:
These were pulled from goutils to drop the `github.com/pkg/errors` package. Unfortunately there was no way to control how those errors get outputed with zap's json encoder.

Reply:
Nice; and I'm all for putting these functions in the `rdk` anyway, seems like a better place.